src/test/scala/vsys/blockchain/state/contract/vswap/ExecuteVSwapValidDiffTest.scala

```diff
@@ -0,0 +1,420 @@
+      createABLiquidityTokenAndInitVSwap(
+        1000, // totalSupplyA
+        1, // unityA
+        1000, // issueAmountA
+        1000, // totalSupplyB
+        1, //unityB
+        1000, // issueAmountB
+        1000, // liquidityTotalSupply
+        1000, // liquidityUnity
+        9, // minimumLiquidity
+        100, // tokenADepositAmount
+        100) // tokenBDepositAmount
+    // set swap
+    setSwap <- setSwapVSwapGen(master, regVSwap.contractId, 10L, 10L, attach, fee + 10000000000L, ts + 10)
+  } yield (genesis, genesis2, regTokenA, regTokenB, regLiquidity, regVSwap, issueTokenA, issueTokenB, issueLiquidity, depositA,
+    depositB, depositLiquidity, setSwap)
+
+  property("vswap able to setswap") {
+    forAll(preconditionsAndVSwapSetSwap) { case (genesis: GenesisTransaction, genesis2: GenesisTransaction, regTokenA: RegisterContractTransaction,
+    regTokenB: RegisterContractTransaction, regLiquidity: RegisterContractTransaction, regVSwap: RegisterContractTransaction, issueTokenA: ExecuteContractFunctionTransaction,
+    issueTokenB: ExecuteContractFunctionTransaction, issueLiquidity: ExecuteContractFunctionTransaction, depositA: ExecuteContractFunctionTransaction, depositB: ExecuteContractFunctionTransaction,
+    depositLiquidity: ExecuteContractFunctionTransaction, setSwap: ExecuteContractFunctionTransaction) =>
+      assertDiffAndStateCorrectBlockTime(Seq(TestBlock.create(genesis.timestamp, Seq(genesis, genesis2)), TestBlock.create(depositLiquidity.timestamp, Seq(regTokenA,
+        regTokenB, regLiquidity, regVSwap, issueTokenA, issueTokenB, issueLiquidity, depositA, depositB, depositLiquidity))),
+        TestBlock.createWithTxStatus(setSwap.timestamp, Seq(setSwap), TransactionStatus.Success)) { (blockDiff, newState) =>
+        blockDiff.txsDiff.txStatus shouldBe TransactionStatus.Success
+
+        val master = setSwap.proofs.firstCurveProof.explicitGet().publicKey
+        val vswapContractId = regVSwap.contractId.bytes
+        val tokenABalanceKey = ByteStr(Bytes.concat(vswapContractId.arr, Array(0.toByte), DataEntry(master.toAddress.bytes.arr, DataType.Address).bytes)) //tokenAReservedStateVar
+        newState.contractNumInfo(tokenABalanceKey) shouldEqual 90L
+        val tokenBBalanceKey = ByteStr(Bytes.concat(vswapContractId.arr, Array(1.toByte), DataEntry(master.toAddress.bytes.arr, DataType.Address).bytes)) //tokenAReservedStateVar
+        newState.contractNumInfo(tokenBBalanceKey) shouldEqual 90L
+        val tokenAReservedKey = ByteStr(Bytes.concat(vswapContractId.arr, Array(6.toByte))) //tokenAReservedStateVar
+        newState.contractNumInfo(tokenAReservedKey) shouldEqual 10L
+        val tokenBReservedKey = ByteStr(Bytes.concat(vswapContractId.arr, Array(7.toByte))) //tokenAReservedStateVar
+        newState.contractNumInfo(tokenBReservedKey) shouldEqual 10L
```

Comment:
How about the user's liquidity balance?

Reply:
liquidity token left, total supply...
And since the key is common used here, we can write a function which input contract id to generate all keys in one line.
`(swapStatusKey, minimuLiquidityKey, tokenAReservedKey, tokenBReservedKey, totalSupplyKey, liquidityTokenLeftKey) = genSwapStateKeys(vswapContractId)`
and 
`(masterTokenABalanceKey, masterTokenBBalanceKey, masterLiquidityBalanceKey) = genUserStateMapKeys(contractId, master)`